### PR TITLE
Adding ColorPicker widget

### PR DESCRIPTION
**Pull request by SylvainCorlay**
Using the native browser color picker.

### CODE REVIEW

**Comment by minrk** [comment]
Travis is showing a SyntaxError. Typos?

**Comment by SylvainCorlay** [comment]
I see the same error on master.

**Comment by minrk** [comment]
Thanks. Seems strange that a commit that changes no javascript would introduce a Javascript SyntaxError, but I will investigate. @jdfreder any hints for finding *where* the SyntaxError occurs, or what the error actually is?

**Comment by jdfreder** [comment]
Darn, just looked at the Travis output, that's weird.  My guess is that it's a problem with one of the requirejs imports.  @minrk printing the error correctly is one of those N-contexts issues, somewhere I'm probably doing `String(Error instance)` which is not providing enough information.  The best way to go about this one is to reproduce the error as a user.  With so many asynchronous things happening at once the order of events isn't always reliable; however, it does look like the error is getting thrown at the beginningish of page/widgets load.

**Comment by minrk** [comment]
jupyter/notebook#472 is not related, since these tests are running on Travis with current stable of the notebook (4.0.5), prior to that change. I can also verify that the tests pass when using current master of the notebook. The problem is that one of the patches on 4.0.5 was incomplete. I just cut 4.0.6 with the fixed patch and everything's green here.

**Comment by SylvainCorlay** [comment]
Sorry for being misleading here - and thanks for solving the issue.

File: ipywidgets/static/widgets/js/widget_color.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+     * From a valid html color (named color, 6-digits or 3-digits hex format)
+     * return a 6-digits hexadecimal color #rrggbb.
+     */
+    function color2hex(color) {
+        return named_colors[color.toLowerCase()] || rgb2hex(color);
+    };
+
+    function rgb2hex(rgb) {
+        if (rgb.search("rgb") == -1) {
+            return rgb;
```
Comment (SylvainCorlay) [inline ipywidgets/static/widgets/js/widget_color.js]: I still need to handle the convertion `#rgb` to `#rrggbb`

File: ipywidgets/static/widgets/js/widget_color.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+            function hex(x) {
+                return ("0" + parseInt(x).toString(16)).slice(-2);
+            }
+            return "#" + hex(rgb[1]) + hex(rgb[2]) + hex(rgb[3]);
+        }
+    };
+
+    return {
+        ColorPicker: ColorPicker,
+        color2hex: color2hex,
```
Comment (jdfreder) [inline ipywidgets/static/widgets/js/widget_color.js]: Want to export rgb2hex too?

File: ipywidgets/static/widgets/js/widget_color.js (hunk 1 above)
Comment (jdfreder) [inline ipywidgets/static/widgets/js/widget_color.js]: What color types does the `Color` trait validate?

File: ipywidgets/static/widgets/js/widget_color.js (hunk 2 above)
Comment (SylvainCorlay) [inline ipywidgets/static/widgets/js/widget_color.js]: I should only export color2hex. And have a hex3/hex6 distinction.

File: ipywidgets/static/widgets/js/widget_color.js (hunk 1 above)
Comment (SylvainCorlay) [inline ipywidgets/static/widgets/js/widget_color.js]: I think both. ( I need to check. ) plus the named colors.

File: ipywidgets/static/widgets/js/widget_color.js (hunk 2 above)
Comment (jdfreder) [inline ipywidgets/static/widgets/js/widget_color.js]: I'm confused, what makes color2hex > rgb2hex ?  Why export one and not the other?   Or instead, not export anything but the widget.

File: ipywidgets/static/widgets/js/widget_color.js (hunk 1 above)
Comment (jdfreder) [inline ipywidgets/static/widgets/js/widget_color.js]: Yeah, whatever `Color` validates, this widget should definitely support.  No more, no less.

File: ipywidgets/static/widgets/js/widget_color.js (hunk 2 above)
Comment (SylvainCorlay) [inline ipywidgets/static/widgets/js/widget_color.js]: color2hex knows about named colors.

Comment (jdfreder) [inline ipywidgets/static/widgets/js/widget_color.js]: Ah sorry! I see now, color2hex is a superset of rgb2hex.  That makes sense

**Comment by SylvainCorlay** [comment]
This is ready.

**Comment by jdfreder** [comment]
Thanks!

**Pull request merged**